app/models/goal.py

```diff
@@ -1,6 +1,21 @@
 from flask import current_app
+from sqlalchemy.orm import relationship
 from app import db
+from app.models.task import Task
+from sqlalchemy import Table, Column, Integer, ForeignKey
+from sqlalchemy.orm import relationship
+from sqlalchemy.ext.declarative import declarative_base
```

Comment:
hm... `Table`, `Column`, etc. should all be coming from your `db` instance of sqlalchemy from `__init__.py`. So I don't think you need the sqlalchemy imports in this file as it's a little redundant.